### ISSUE TITLE
Should the template setup localizations?

### ISSUE THREAD

**Issue by brianegan** [brianegan/new_flutter_template#4]
Should the template include the necessary work to set up localizations? This includes adding an l10n.yaml, en.arb, modify the pubspec.yaml to generate localization, and use the AppLocalizations for Strings throughout the app. 

This might be overkill for intermediate developers or it might be a healthy nudge in the right direction. What are your thoughts?

  1. No localizations
  2. Placeholder that links to more information
  2. Localize only the title of the app
  3. All strings localized

Background: https://flutter.dev/docs/development/accessibility-and-localization/internationalization

**Comment by brianegan** [comment]
No

**Comment by brianegan** [comment]
Yes

**Comment by brianegan** [comment]
Placeholder that links to more information

**Comment by brianegan** [comment]
Yes

**Comment by orestesgaolin** [comment]
Yes, I'm in favor of new localization support by using basic localization delegate and translating _all strings_ to English as default. It might be necessary to explain `flutter.generate` property in `pubspec.yaml` in more detail.

**Comment by chimon2000** [comment]
Yes, I think localization by default is a great feature!

**Comment by RobertBrunhage** [comment]
I am a bit on the fence with this. Sure it's nice to have it on a project but this also adds more things a novice need to think about right of the bat. But I am very much in favor of adding some kind of documentation stating how and why you would implement it.

I may have missed it but looking at other starting projects from other frameworks, localizations is a typical thing that is not implemented in the beginning.

**Comment by alestiago** [comment]
Would appreciate it. Both 3 and 4 sound great to me.

**Comment by mhadaily** [comment]
Having a quick setup for localization is very nice, to have a better example I belive we should go for 4- All strings localized

**Comment by burhanrashid52** [comment]
I would go with options 2. And also agrees with @RobertBrunhage comment of "adding more things". Developers who are building apps for countries that do not have English as their primary language will tend to have this feature to be included in the template.
Setting up localization requires a lot of boilerplate code and very hard to understand at first glance. IMO for beginners, it will be too much to learn.
